src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -1543,6 +1603,114 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 
 		navigationController?.present(alert, animated: true)
 	}
+	
+	private func showSRSFlowAlert(
+		for srsFlowAlert: SRSFlowAlert,
+		isLoading: @escaping CompletionBool
+	) {
+		switch srsFlowAlert {
+		case let .preconditionFailed(srsFlowError):
+			showSRSFlowPreconditionAlert(for: srsFlowError)
+		case let .consent(srsFlowConsent):
+			showSRSFlowConsentAlert(for: srsFlowConsent, isLoading: isLoading)
+		case let .error(error):
+			showSRSFlowErrorAlert(for: error, isLoading: isLoading)
+		}
+	}
+	
+	private func showSRSFlowConsentAlert(
+		for consent: SRSFlowAlert.Consent,
+		isLoading: @escaping CompletionBool
+	) {
+		var alert: UIAlertController!
+		switch consent {
+		case .cancelWarnOthers(let viewController):
+			alert = UIAlertController(
+				title: AppStrings.ExposureSubmission.SRSTestTypeSelection.warnProcessCancelAlertTitle,
+				message: AppStrings.ExposureSubmission.SRSTestTypeSelection.warnProcessCancelAlertMessage,
+				preferredStyle: .alert
+			)
+			
+			let continueAction = UIAlertAction(
+				title: AppStrings.ExposureSubmission.SRSTestTypeSelection.warnProcessCancelAlertActionContinue,
+				style: .default
+			)
+			alert.addAction(continueAction)
+			
+			alert.addAction(UIAlertAction(
+				title: AppStrings.ExposureSubmission.SRSTestTypeSelection.warnProcessCancelAlertActionCancel,
+				style: .cancel,
+				handler: { [weak self] _ in
+					self?.navigationController?.dismissAllModalViewControllers(animated: true)
+				}
+			))
+			
+			alert.preferredAction = continueAction
+			
+			viewController.present(alert, animated: true)
+			
+		case .confirmWarnOthers(let viewController):
+			alert = UIAlertController(
+				title: AppStrings.SRSConfirmWarnOthersAlert.title,
+				message: AppStrings.SRSConfirmWarnOthersAlert.message,
+				preferredStyle: .alert
+			)
+			
+			let confirmAction = UIAlertAction(
+				title: AppStrings.SRSConfirmWarnOthersAlert.actionConfirm,
+				style: .default,
+				handler: { [weak self] _ in
+					self?.submitSRSExposure(showSubmissionSuccess: true, isLoading: isLoading)
+				}
+			)
+			alert.addAction(confirmAction)
+			
+			alert.addAction(UIAlertAction(
+				title: AppStrings.SRSConfirmWarnOthersAlert.actionCancel,
+				style: .cancel,
+				handler: { [weak self] _ in
+					self?.navigationController?.dismissAllModalViewControllers(animated: true)
+				}
+			))
+			
+			alert.preferredAction = confirmAction
+			
+			viewController.present(alert, animated: true)
```

Comment:
That was my first approach, but unfortunately that was not possible, because nav controller is already presenting a VC. Wo don't know which VC is the top most one. I also tried some SOF approaches, but the simplest way is, to give the VC as an associated value to the type.